### ISSUE TITLE
someone explain to me why I'm unable to update the cache after removing an item?

### ISSUE THREAD

**Issue by Deepak-RefactorQ** [reduxjs/redux-toolkit#4144]
`
endpoints: (builder) => ({
  getCart: builder.query({
    query: () => ({
      url: `/cart`,
    }),
    transformResponse: (response) => response.result,
    providesTags: ["Cart"],
  }),
  removeFromCart: builder.mutation({
    query: (ids) => ({
      url: "/cart",
      method: "DELETE",
      params: {
        ids: ids.join(","),
      },
    }),
    transformResponse: (response) => response.result,
    async onQueryStarted(args, { dispatch, queryFulfilled }) {
      try {
        const { data: updatedCart } = await queryFulfilled;

        dispatch(updateCartItems(updatedCart));

        const patchResult = dispatch(
          cartApi.util.updateQueryData("getCart", null, (draft) => {
            Object.assign(draft, updatedCart);
          })
        );
      } catch (error) {
        console.log(error);
      }
    },
    invalidatesTags: ["Cart-Count"],
  }),
});

`
Its working fine with the "update" query.

**Comment by juliengbt** [comment]
Have you considered using `undefined` instead of `null` in the `updateQueryData` call

**Comment by EskiMojo14** [comment]
yeah, the argument passed needs to match - no argument is equivalent to undefined, not null.
